File: src/main/java/org/taktik/icure/entities/embed/CareTeamMember.java
package org.taktik.icure.entities.embed;

import org.taktik.icure.entities.base.CodeStub;

import java.io.Serializable;
import java.util.Objects;

public class CareTeamMember implements Serializable {
    private String id;
    private CareTeamMemberType careTeamMemberType;
    private String healthcarePartyId;
    private CodeStub quality;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public CareTeamMemberType getCareTeamMemberType() {
        return careTeamMemberType;
    }

    public void setCareTeamMemberType(CareTeamMemberType careTeamMemberType) {
        this.careTeamMemberType = careTeamMemberType;
    }

    public String getHealthcarePartyId() {
        return healthcarePartyId;
    }

    public void setHealthcarePartyId(String healthcarePartyId) {
        this.healthcarePartyId = healthcarePartyId;
    }

    public CodeStub getQuality() {
        return quality;
    }

    public void setQuality(CodeStub quality) {
        this.quality = quality;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof CareTeamMember)) return false;
        CareTeamMember that = (CareTeamMember) o;
        return Objects.equals(id, that.id) &&
                careTeamMemberType == that.careTeamMemberType &&
                Objects.equals(healthcarePartyId, that.healthcarePartyId) &&
                Objects.equals(quality, that.quality);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, careTeamMemberType, healthcarePartyId, quality);
    }
}

File: src/main/kotlin/org/taktik/icure/entities/embed/CareTeamMember.kt
package org.taktik.icure.entities.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.pozo.KotlinBuilder
import org.taktik.icure.entities.base.CodeStub
import org.taktik.icure.entities.base.Identifiable
import org.taktik.icure.utils.DynamicInitializer
import org.taktik.icure.utils.invoke
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class CareTeamMember(
        @JsonProperty("_id") override val id: String,
        val careTeamMemberType: CareTeamMemberType? = null,
        val healthcarePartyId: String? = null,
        val quality: CodeStub? = null,
        override val encryptedSelf: String? = null
) : Encrypted, Serializable, Identifiable<String> {
    companion object : DynamicInitializer<CareTeamMember>

    fun merge(other: CareTeamMember) = CareTeamMember(args = this.solveConflictsWith(other))
    fun solveConflictsWith(other: CareTeamMember) = super.solveConflictsWith(other) + mapOf(
            "id" to (this.id),
            "careTeamMemberType" to (this.careTeamMemberType ?: other.careTeamMemberType),
            "healthcarePartyId" to (this.healthcarePartyId ?: other.healthcarePartyId),
            "quality" to (this.quality ?: other.quality)
    )
}
